### PR TITLE
Create Default Procs on-the-fly when first called

### PR DESCRIPTION
**Pull request by jhugg**
I may try to make it easier to make shim Procedures/Statements, but this is ready for review.

### CODE REVIEW

File: src/frontend/org/voltdb/DefaultProcedureManager.java [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+
+    public Procedure checkForDefaultProcedure(String name) {
+        return m_defaultProcMap.get(name.toLowerCase());
+    }
+
+    private void build() {
+        for (Table table : m_db.getTables()) {
+            String prefix = table.getTypeName() + '.';
+
+            // skip export tables XXX why no insert?
```
Comment (mikeismert) [inline src/frontend/org/voltdb/DefaultProcedureManager.java]: Are these cut-and-pasted comments or new XXX wonderings that you added?

File: src/frontend/org/voltdb/DefaultProcedureManager.java [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+            }
+        case "upsert":
+            if (defaultProc.getSinglepartition()) {
+                return generateCrudUpsert(table, partitionColumn);
+            }
+            else {
+                return generateCrudReplicatedUpsert(table, pkey);
+            }
+        default:
+            throw new RuntimeException("Invlid input to default proc SQL generator.");
```
Comment (mikeismert) [inline src/frontend/org/voltdb/DefaultProcedureManager.java]: Typo: 'Invalid'

File: src/frontend/org/voltdb/DefaultProcedureManager.java [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+            addShimProcedure(prefix + "update", table, pkey, true, columnCount + pkeyPartitionIndex, partitioncolumn, false);
+            // upsert partitions like a regular insert
+            addShimProcedure(prefix + "upsert", table, null, true, partitionIndex, partitioncolumn, false);
+        }
+    }
+
+    public String sqlForDefaultProc(Procedure defaultProc) {
+        String name = defaultProc.getClassname();
+        String[] parts = name.split("\\.");
+        String action = parts[1];
```
Comment (mikeismert) [inline src/frontend/org/voltdb/DefaultProcedureManager.java]: toLowerCase() ?

File: src/frontend/org/voltdb/DefaultProcedureManager.java [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+        Constraint pkey = null;
+        while (it.hasNext()) {
+            Constraint constraint = it.next();
+            if (constraint.getType() == ConstraintType.PRIMARY_KEY.getValue()) {
+                pkey = constraint;
+                break;
+            }
+        }
+
+        switch(action) {
```
Comment (mikeismert) [inline src/frontend/org/voltdb/DefaultProcedureManager.java]: or here?

File: src/frontend/org/voltdb/DefaultProcedureManager.java (hunk 3 above)
Comment (mikeismert) [inline src/frontend/org/voltdb/DefaultProcedureManager.java]: I guess we already know we only make these lower?

File: src/frontend/org/voltdb/DefaultProcedureManager.java (hunk 1 above)
Comment (jhugg) [inline src/frontend/org/voltdb/DefaultProcedureManager.java]: I think I added those.

File: src/frontend/org/voltdb/planner/ActivePlanRepository.java [hunk 5]
```diff
@@ -85,7 +85,7 @@ public static String getStmtTextForPlanHash(byte[] planHash) {
             frag = m_plansByHash.get(key);
         }
         assert(frag != null);
-        assert(frag.stmtText != null);
+        //assert(frag.stmtText != null);
```
Comment (mikeismert) [inline src/frontend/org/voltdb/planner/ActivePlanRepository.java]: Dead code?  Purge!

**Pull request merged**